tools/djxl_ng_main.cc

```diff
@@ -0,0 +1,299 @@
+DECLARE_bool(helpshort);
+
+DEFINE_int64(num_reps, 1,  // TODO(firsching): Clarify meaning of this
+                           // docstring. Is this simply for benchmarking?
+             "How many times to decompress.");
+
+DEFINE_int64(num_threads, 0,
+             // TODO(firsching): Sync with team about changed meaning of 0 -
+             // was: No multithreaded workers. Is: use default number.
+             "Number of worker threads (0 == use machine default).");
+
+// TODO(firsching): wire this up.
+DEFINE_bool(print_profile, false, "Print timing information before exiting.");
+
+// TODO(firsching): wire this up.
+DEFINE_int32(bits_per_sample, 0, "0 = original (input) bit depth");
+
+// TODO(firsching): wire this up.
+DEFINE_bool(
+    tone_map, true,
+    "tone map the image to the luminance range indicated by --display_nits "
+    "instead of performing a naive 0-1 -> 0-1 conversion");
+
+// TODO(firsching): wire this up.
+DEFINE_string(display_nits, "0.f-255.",
+              "luminance range of the display to which to "
+              "tone-map; the lower bound can be omitted");
+
+// TODO(firsching): wire this up.
+DEFINE_double(preserve_saturation, 0.1,
+              "with --tone_map, how much to favor saturation over luminance");
+
+// TODO(firsching): wire this up.
+DEFINE_string(color_space, "RGB_D65_SRG_Rel_Lin",
+              "defaults to original (input) color space");
+
+// TODO(firsching): wire this up.
+DEFINE_uint32(downsampling, 0,
+              "maximum permissible downsampling factor (values "
+              "greater than 16 will return the LQIP if available");
+
+// TODO(firsching): wire this up.
+DEFINE_bool(allow_partial_files, false, "allow decoding of truncated files");
+
+// TODO(firsching): wire this up.
+DEFINE_bool(allow_more_progressive_steps, false,
+            "allow decoding more progressive steps in truncated "
+            "files. No effect without --allow_partial_files");
+
+#if JPEGXL_ENABLE_JPEG
+// TODO(firsching): wire this up.
+DEFINE_bool(
+    pixels_to_jpeg, false,
+    "By default, if the input JPEG XL contains a recompressed JPEG file, "
+    "djxl "
+    "reconstructs the exact original JPEG file. This flag causes the decoder "
+    "to instead decode the image to pixels and encode a new (lossy) JPEG. "
+    "The output file if provided must be a .jpg or .jpeg file.");
+
+// TODO(firsching): wire this up.
+DEFINE_uint32(jpeg_quality, 95,
+              "JPEG output quality. Setting an output quality "
+              "implies --pixels_to_jpeg.");
+#endif
+
+#if JPEGXL_ENABLE_SJPEG
+// TODO(firsching): wire this up.
+DEFINE_bool(use_sjpeg, false, "use sjpeg instead of libjpeg for JPEG output");
+#endif
+
+// TODO(firsching): wire this up.
+DEFINE_bool(print_read_bytes, false, "print total number of decoded bytes");
+
+// TODO(firsching): wire this up.
+DEFINE_bool(quiet, false, "silence output (except for errors)");
+
+bool WriteFile(const char* filename, const std::vector<uint8_t> bytes) {
```

Comment:
I think the vector wants to be a const reference

Reply:
yes, of course